packages/core/src/node-execution-context/helpers/binary-helpers.ts

```diff
@@ -0,0 +1,148 @@
+import FileType from 'file-type';
+import { IncomingMessage } from 'http';
+import MimeTypes from 'mime-types';
+import { ApplicationError, fileTypeFromMimeType } from 'n8n-workflow';
+import type {
+	BinaryHelperFunctions,
+	IWorkflowExecuteAdditionalData,
+	Workflow,
+	IBinaryData,
+} from 'n8n-workflow';
+import path from 'path';
+import type { Readable } from 'stream';
+import Container from 'typedi';
+
+import { BinaryDataService } from '@/BinaryData/BinaryData.service';
+import { binaryToBuffer } from '@/BinaryData/utils';
+// eslint-disable-next-line import/no-cycle
+import { binaryToString } from '@/NodeExecuteFunctions';
+
+export class BinaryHelpers {
```

Comment:
Can we have `BinaryFunctions`, `SchedulingFunctions`, and `RequestFunctions`? And they return `ExportedBinaryFunctions`, `ExportedSchedulingFunctions`, `ExportedRequestFunctions`?